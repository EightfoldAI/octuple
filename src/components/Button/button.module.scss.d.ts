declare namespace ButtonModuleScssModule {
    export interface IButtonModuleScss {
        body1: string;
        body2: string;
        body3: string;
        button: string;
        'button-default': string;
        'button-neutral': string;
        'button-padding-1': string;
        'button-padding-1-default': string;
        'button-padding-1-neutral': string;
        'button-padding-2': string;
        'button-padding-2-default': string;
        'button-padding-2-neutral': string;
        'button-padding-3': string;
        'button-padding-3-default': string;
        'button-padding-3-neutral': string;
        'button-primary': string;
        'button-primary-disruptive': string;
        'button-secondary': string;
        'button-secondary-disruptive': string;
        'button-spacer-1': string;
        'button-spacer-2': string;
        'button-spacer-3': string;
        button1: string;
        button2: string;
        button3: string;
        caption: string;
        'chevron-icon': string;
        'chevron-spacer': string;
        'custom-padding-vertical': string;
        disabled: string;
        display1: string;
        display2: string;
        display3: string;
        display4: string;
        'divider-line': string;
        'flex-structure-horizontal': string;
        'flex-structure-vertical': string;
        'full-height': string;
        header1: string;
        header2: string;
        header3: string;
        header4: string;
        header5: string;
        header6: string;
        icon: string;
        'icon-1-material': string;
        'icon-2-material': string;
        'icon-3-material': string;
        'icon-4-material': string;
        'icon-material-l': string;
        'icon-material-m': string;
        'icon-material-s': string;
        'icon-material-xs': string;
        l: string;
        'logo-text': string;
        m: string;
        mappings: string;
        'menu-padding-1': string;
        'menu-padding-2': string;
        'menu-padding-3': string;
        names: string;
        'octuple-menu': string;
        'open-sans': string;
        'pre-display': string;
        s: string;
        semibold: string;
        sourceRoot: string;
        sources: string;
        sourcesContent: string;
        'space-m': string;
        spacer: string;
        subtitle1: string;
        subtitle2: string;
        version: string;
        xl: string;
        xxl: string;
        xxxl: string;
    }
}

declare const ButtonModuleScssModule: ButtonModuleScssModule.IButtonModuleScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: ButtonModuleScssModule.IButtonModuleScss;
};

export = ButtonModuleScssModule;
