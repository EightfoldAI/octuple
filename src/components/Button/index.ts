export * from './Button.types';
export * from './BaseButton';
export * from './DefaultButton/DefaultButton';
export * from './NeutralButton/NeutralButton';
export * from './PrimaryButton/PrimaryButton';
export * from './SecondaryButton/SecondaryButton';
export * from './SplitButton/SplitButton';
export * from './TwoStateButton/TwoStateButton';
