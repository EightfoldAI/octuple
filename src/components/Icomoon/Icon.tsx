import React, { FC, useContext } from 'react';
import { mergeClasses } from '../../shared/utilities';
import { IcomoonProps, IcomoonSize } from './';
import IcomoonReact from 'icomoon-react';

import styles from './icon.module.scss';

const IconSetContext = React.createContext<object | null>(null);

export const IcomoonProvider = IconSetContext.Provider;

export const Icomoon: FC<IcomoonProps> = ({
    ariaHidden = false,
    classNames,
    color,
    iconName,
    id,
    role = 'presentation',
    size = IcomoonSize.Medium,
    'data-test-id': dataTestId,
}) => {
    const iconClassNames: string = mergeClasses([
        classNames,
        styles.iconWrapper,
    ]);

    return (
        <span
            data-test-id={dataTestId}
            aria-hidden={ariaHidden}
            className={iconClassNames}
            id={id}
            role={role}
        >
            <IconSetContext.Consumer>
                {(iconSet) => (
                    <IcomoonReact
                        iconSet={iconSet}
                        icon={iconName}
                        size={size}
                        color={color}
                    />
                )}
            </IconSetContext.Consumer>
        </span>
    );
};
