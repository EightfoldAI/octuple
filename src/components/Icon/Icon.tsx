'use client';

import React, { FC } from 'react';
import IcomoonReact from 'icomoon-react';

import { mergeClasses } from '../../shared/utilities';
import { Icon as MdiIcon } from '@mdi/react';
import { IconProps, IconSize } from './index';

import styles from './icon.module.scss';
import { useConfig } from '../ConfigProvider';

export const Icon: FC<IconProps> = ({
  ariaHidden = false,
  classNames,
  color,
  description,
  horizontal,
  id,
  path,
  role = 'presentation',
  rotate,
  size = IconSize.Medium,
  spin,
  style,
  title,
  vertical,
  'data-test-id': dataTestId,
  icomoonIconName,
}) => {
  const { icomoonIconSet } = useConfig();

  const iconClassNames: string = mergeClasses([classNames, styles.iconWrapper]);

  if (!icomoonIconName && !path) {
    return null;
  }

  const iconComponent = icomoonIconName ? (
    <IcomoonReact
      iconSet={icomoonIconSet}
      size={size}
      color={color}
      icon={icomoonIconName}
      role="presentation"
    />
  ) : (
    <MdiIcon
      color={color}
      description={description}
      horizontal={horizontal}
      path={path}
      rotate={rotate}
      size={size}
      title={title}
      vertical={vertical}
      spin={spin}
    />
  );

  return (
    <span
      data-test-id={dataTestId}
      aria-hidden={ariaHidden}
      className={iconClassNames}
      id={id}
      role={role}
      style={style ? style : null}
    >
      {iconComponent}
    </span>
  );
};

Icon.displayName = 'Icon';
