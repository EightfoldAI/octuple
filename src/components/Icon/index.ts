export * from './Icon';
export * from './Icon.types';
export * from './mdi';
