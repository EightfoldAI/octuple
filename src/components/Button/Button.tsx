import React, { FC, Ref, useContext, useEffect, useRef, useState } from 'react';
import DisabledContext, { Disabled } from '../ConfigProvider/DisabledContext';
import { SizeContext, Size } from '../ConfigProvider';
import {
  ButtonIconAlign,
  ButtonProps,
  ButtonShape,
  ButtonSize,
  ButtonTextAlign,
  ButtonVariant,
  ButtonWidth,
  SplitButton,
} from '.';
import { Icon, IconSize } from '../Icon';
import { Badge } from '../Badge';
import { InnerNudge, NudgeAnimation, NudgeProps } from './Nudge';
import { Breakpoints, useMatchMedia } from '../../hooks/useMatchMedia';
import { mergeClasses } from '../../shared/utilities';
import { Loader, LoaderSize } from '../Loader';
import { useCanvasDirection } from '../../hooks/useCanvasDirection';
import { useMergedRefs } from '../../hooks/useMergedRefs';
import { useNudge } from './Nudge/Hooks/useNudge';

import styles from './button.module.scss';

export const Button: FC<ButtonProps> = React.forwardRef(
  (props: ButtonProps, ref: Ref<HTMLButtonElement>) => {
    const {
      alignIcon = ButtonIconAlign.Left,
      alignText = ButtonTextAlign.Center,
      allowDisabledFocus = false,
      ariaLabel,
      buttonWidth = ButtonWidth.fitContent,
      checked = false,
      classNames,
      configContextProps = {
        noDisabledContext: false,
        noSizeContext: false,
      },
      counter,
      disabled = false,
      disruptive = false,
      dropShadow = false,
      floatingButtonProps,
      nudgeProps: defaultNudgeProps,
      htmlType,
      iconProps,
      id,
      loading = false,
      onClick,
      onContextMenu,
      prefixIconProps,
      shape = ButtonShape.Pill,
      size = ButtonSize.Medium,
      split,
      splitButtonChecked = false,
      splitButtonProps,
      style,
      text,
      toggle,
      transparent = false,
      type, // TODO: Remove in Octuple v3.0.0 and use `variant` only.
      variant = ButtonVariant.Default,
      ...rest
    } = props;
    const largeScreenActive: boolean = useMatchMedia(Breakpoints.Large);
    const mediumScreenActive: boolean = useMatchMedia(Breakpoints.Medium);
    const smallScreenActive: boolean = useMatchMedia(Breakpoints.Small);
    const xSmallScreenActive: boolean = useMatchMedia(Breakpoints.XSmall);

    const internalRef: React.MutableRefObject<HTMLButtonElement> =
      useRef<HTMLButtonElement>(null);

    const mergedRef = useMergedRefs(internalRef, ref);

    const htmlDir: string = useCanvasDirection();

    const contextuallyDisabled: Disabled = useContext(DisabledContext);
    const mergedDisabled: boolean = configContextProps.noDisabledContext
      ? disabled
      : contextuallyDisabled || disabled;

    const contextuallySized: Size = useContext(SizeContext);
    const mergedSize = configContextProps.noSizeContext
      ? size
      : contextuallySized || size;

    const counterExists: boolean = !!counter;
    const iconExists: boolean = !!iconProps;
    const prefixIconExists: boolean = !!prefixIconProps;
    const textExists: boolean = !!text;

    const [nudgeProps, setNudgeProps] = useState<NudgeProps>(defaultNudgeProps);
    const innerNudgeRef: React.MutableRefObject<HTMLSpanElement> =
      useRef<HTMLSpanElement>(null);

    useEffect(() => {
      setNudgeProps(
        props.nudgeProps
          ? props.nudgeProps
          : {
              animation: NudgeAnimation.Background,
              delay: 2000,
              iterations: 1,
              enabled: false,
            }
      );
    }, [nudgeProps?.enabled]);

    useNudge(disruptive, nudgeProps, [internalRef, innerNudgeRef], styles);

    // TODO: Remove in Octuple v3.0.0 and use `variant` only.
    // For now, if `type` has a value use it, else use `variant`.
    const mergedVariant: ButtonVariant = !!type ? type : variant;

    const buttonSharedClassNames: string = mergeClasses([
      classNames,
      styles.button,
      { [styles.buttonDefault]: mergedVariant === ButtonVariant.Default },
      { [styles.buttonNeutral]: mergedVariant === ButtonVariant.Neutral },
      { [styles.buttonPrimary]: mergedVariant === ButtonVariant.Primary },
      { [styles.buttonSecondary]: mergedVariant === ButtonVariant.Secondary },
      { [styles.buttonSystemUi]: mergedVariant === ButtonVariant.SystemUI },
      {
        [styles.buttonDisruptive]:
          disruptive && mergedVariant === ButtonVariant.Default,
      },
      {
        [styles.buttonPrimaryDisruptive]:
          disruptive && mergedVariant === ButtonVariant.Primary,
      },
      {
        [styles.buttonSecondaryDisruptive]:
          disruptive && mergedVariant === ButtonVariant.Secondary,
      },
      {
        [styles.transparent]:
          transparent && mergedVariant === ButtonVariant.SystemUI,
      },
      {
        [styles.buttonSmall]:
          mergedSize === ButtonSize.Flex && largeScreenActive,
      },
      {
        [styles.buttonMedium]:
          mergedSize === ButtonSize.Flex && mediumScreenActive,
      },
      {
        [styles.buttonMedium]:
          mergedSize === ButtonSize.Flex && smallScreenActive,
      },
      {
        [styles.buttonLarge]:
          mergedSize === ButtonSize.Flex && xSmallScreenActive,
      },
      { [styles.buttonLarge]: mergedSize === ButtonSize.Large },
      { [styles.buttonMedium]: mergedSize === ButtonSize.Medium },
      { [styles.buttonSmall]: mergedSize === ButtonSize.Small },
      { [styles.pillShape]: shape === ButtonShape.Pill },
      {
        [styles.roundShape]:
          shape === ButtonShape.Round && !split && !textExists,
      },
      { [styles.dropShadow]: dropShadow },
      { [styles.disabled]: allowDisabledFocus || mergedDisabled },
      { [styles.floating]: floatingButtonProps?.enabled },
      { [styles.buttonRtl]: htmlDir === 'rtl' },
      {
        [styles.buttonConic]:
          !disruptive &&
          nudgeProps?.enabled &&
          nudgeProps?.animation === NudgeAnimation.Conic,
      },
    ]);

    const buttonClassNames: string = mergeClasses([
      buttonSharedClassNames,
      { [styles.buttonStretch]: buttonWidth === ButtonWidth.fill },
      { [styles.splitLeft]: split },
      { [styles.left]: alignText === ButtonTextAlign.Left },
      { [styles.right]: alignText === ButtonTextAlign.Right },
      {
        [styles.iconLeft]: iconExists && alignIcon === ButtonIconAlign.Left,
      },
      {
        [styles.iconRight]: iconExists && alignIcon === ButtonIconAlign.Right,
      },
      { [styles.loading]: loading },
    ]);

    const buttonTextClassNames: string = mergeClasses([
      {
        [styles.buttonTextSmall]:
          mergedSize === ButtonSize.Flex && largeScreenActive,
      },
      {
        [styles.buttonTextMedium]:
          mergedSize === ButtonSize.Flex && mediumScreenActive,
      },
      {
        [styles.buttonTextMedium]:
          mergedSize === ButtonSize.Flex && smallScreenActive,
      },
      {
        [styles.buttonTextLarge]:
          mergedSize === ButtonSize.Flex && xSmallScreenActive,
      },
      { [styles.buttonTextLarge]: mergedSize === ButtonSize.Large },
      { [styles.buttonTextMedium]: mergedSize === ButtonSize.Medium },
      { [styles.buttonTextSmall]: mergedSize === ButtonSize.Small },
    ]);

    const badgeClassNames: string = mergeClasses([
      styles.counter,
      buttonTextClassNames,
    ]);

    const getButtonIconSize = (): IconSize => {
      let iconSize: IconSize;
      if (mergedSize === ButtonSize.Flex && largeScreenActive) {
        iconSize = IconSize.Small;
      } else if (
        mergedSize === ButtonSize.Flex &&
        (mediumScreenActive || smallScreenActive)
      ) {
        iconSize = IconSize.Medium;
      } else if (mergedSize === ButtonSize.Flex && xSmallScreenActive) {
        iconSize = IconSize.Large;
      } else if (mergedSize === ButtonSize.Large) {
        iconSize = IconSize.Large;
      } else if (mergedSize === ButtonSize.Medium) {
        iconSize = IconSize.Medium;
      } else if (mergedSize === ButtonSize.Small) {
        iconSize = IconSize.Small;
      }
      return iconSize;
    };

    const getLoaderSize = (): LoaderSize => {
      let loaderSize: LoaderSize;
      if (size === ButtonSize.Flex && largeScreenActive) {
        loaderSize = LoaderSize.Small;
      } else if (
        size === ButtonSize.Flex &&
        (mediumScreenActive || smallScreenActive)
      ) {
        loaderSize = LoaderSize.Medium;
      } else if (size === ButtonSize.Flex && xSmallScreenActive) {
        loaderSize = LoaderSize.Large;
      } else if (size === ButtonSize.Large) {
        loaderSize = LoaderSize.Large;
      } else if (size === ButtonSize.Medium) {
        loaderSize = LoaderSize.Medium;
      } else if (size === ButtonSize.Small) {
        loaderSize = LoaderSize.Small;
      }
      return loaderSize;
    };

    const getButtonLoader = (): JSX.Element =>
      loading && (
        <Loader
          classNames={styles.loader}
          dotClassNames={styles.loaderDot}
          size={getLoaderSize()}
        />
      );

    const getButtonIcon = (): JSX.Element => (
      <Icon
        {...iconProps}
        classNames={mergeClasses([styles.icon, iconProps.classNames])}
        size={getButtonIconSize()}
      />
    );

    const getPrefixIcon = (): JSX.Element => (
      <Icon
        {...prefixIconProps}
        classNames={mergeClasses([
          styles.icon,
          styles.prefixIcon,
          prefixIconProps.classNames,
        ])}
        size={getButtonIconSize()}
      />
    );

    const getButtonContent = (
      buttonTextClassNames: string,
      text: string
    ): JSX.Element => (
      <span className={buttonTextClassNames}>
        {text ? text : 'Button'}
        {counterExists && <Badge classNames={badgeClassNames}>{counter}</Badge>}
      </span>
    );

    return (
      <>
        <button
          {...rest}
          ref={mergedRef}
          aria-checked={toggle ? !!checked : undefined}
          aria-disabled={mergedDisabled || loading}
          aria-label={ariaLabel}
          aria-pressed={toggle ? !!checked : undefined}
          defaultChecked={checked}
          disabled={(!allowDisabledFocus && mergedDisabled) || loading}
          className={buttonClassNames}
          id={id}
          onClick={!allowDisabledFocus ? onClick : null}
          style={style}
          type={htmlType}
        >
          <InnerNudge
            classNames={mergeClasses([
              styles.innerNudge,
              {
                [styles.conic]: nudgeProps?.animation === NudgeAnimation.Conic,
              },
            ])}
            disruptive={disruptive}
            id={id ? `${id}-nudge` : 'button-nudge'}
            nudgeProps={nudgeProps}
            ref={innerNudgeRef}
            style={style}
          />
          {iconExists && prefixIconExists && !textExists && (
            <span>
              {getButtonIcon()}
              {getPrefixIcon()}
            </span>
          )}
          {iconExists &&
            !textExists &&
            !prefixIconExists &&
            !counterExists &&
            getButtonIcon()}
          {counterExists && !textExists && !loading && !iconExists && (
            <Badge classNames={badgeClassNames}>{counter}</Badge>
          )}
          {iconExists && counterExists && !textExists && !loading && (
            <span>
              {getButtonIcon()}
              <Badge classNames={badgeClassNames}>{counter}</Badge>
              {prefixIconExists && getPrefixIcon()}
            </span>
          )}
          {iconExists && textExists && (
            <span>
              {getButtonIcon()}
              {getButtonContent(buttonTextClassNames, text)}
              {prefixIconExists && getPrefixIcon()}
            </span>
          )}
          {!iconExists && getButtonContent(buttonTextClassNames, text)}
          {getButtonLoader()}
        </button>
        {split && (
          <SplitButton
            {...splitButtonProps}
            classNames={
              buttonSharedClassNames + ' ' + splitButtonProps?.classNames
            }
            checked={splitButtonChecked}
            disruptive={disruptive}
            dropShadow={dropShadow}
            nudgeProps={nudgeProps}
            onClick={
              !splitButtonProps?.allowDisabledFocus ? onContextMenu : null
            }
            shape={shape}
            size={mergedSize}
            split={split}
            variant={mergedVariant}
          />
        )}
      </>
    );
  }
);
