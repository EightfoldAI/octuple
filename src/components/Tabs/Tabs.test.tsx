import React from 'react';
import Enzyme, { mount, ReactWrapper } from 'enzyme';
import { create } from 'react-test-renderer';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import MatchMediaMock from 'jest-matchmedia-mock';
import { Stat, Tabs, Tab, TabSize, TabVariant, TabIconAlign } from './';
import { ButtonShape, ButtonVariant } from '../Button';
import { IconName } from '../Icon';

Enzyme.configure({ adapter: new Adapter() });

let matchMedia: any;

describe('Tabs', () => {
  let wrapper: ReactWrapper;

  const tabs = [1, 2, 3, 4].map((i) => ({
    value: `tab${i}`,
    icon: IconName.mdiCardsHeart,
    label: `Tab ${i}`,
    ariaLabel: `Tab ${i}`,
    ...(i === 4 ? { disabled: true } : {}),
  }));
  const tabsWithBadgeContent = [1, 2, 3, 4].map((i) => ({
    value: `tab${i}`,
    icon: IconName.mdiCardsHeart,
    label: `Tab ${i}`,
    ariaLabel: `Tab ${i}`,
    badgeContent: i,
    ...(i === 4 ? { disabled: true } : {}),
  }));
  const tabsWithButtons = [1, 2, 3, 4].map((i) => ({
    buttonProps: {
      ariaLabel: 'Send reminder',
      iconProps: { path: IconName.mdiBellOutline },
      onClick: (e: React.MouseEvent<HTMLButtonElement>) => {
        e.stopPropagation();
        console.log('clicked');
      },
      shape: ButtonShape.Round,
      variant: ButtonVariant.Neutral,
    },
    value: `tab${i}`,
    icon: IconName.mdiCardsHeart,
    label: `Tab ${i}`,
    ariaLabel: `Tab ${i}`,
    ...(i === 4 ? { disabled: true } : {}),
  }));
  const tabClick = jest.fn();

  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });
  afterEach(() => {
    matchMedia.clear();
  });

  beforeEach(() => {
    wrapper = mount(
      <Tabs onChange={tabClick} value={'tab1'}>
        {tabs.map((tab) => (
          <Tab key={tab.value} {...tab} />
        ))}
      </Tabs>
    );
  });

  test('tabs snapshot', () => {
    expect(
      create(
        <Tabs onChange={tabClick} value={'tab1'}>
          {tabs.map((tab) => (
            <Tab key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('tabs mount', () => {
    expect(wrapper.find('.tab').at(0).hasClass('active')).toEqual(true);
  });

  test('tabs click', () => {
    wrapper.find('.tab').at(2).simulate('click');
    expect(tabClick).toHaveBeenCalledTimes(1);
    expect(tabClick).toHaveBeenCalledWith('tab3', expect.any(Object));
  });

  test('setting value changes tab', async () => {
    wrapper = mount(
      <Tabs onChange={tabClick} value={'tab3'}>
        {tabs.map((tab) => (
          <Tab key={tab.value} {...tab} />
        ))}
      </Tabs>
    );
    expect(wrapper.find('.tab').at(2).hasClass('active')).toEqual(true);
  });

  test('tabs badgeContent', () => {
    expect(
      create(
        <Tabs
          alignIcon={TabIconAlign.Start}
          onChange={tabClick}
          value={'tab1'}
          variant={TabVariant.pill}
        >
          {tabsWithBadgeContent.map((tab) => (
            <Tab key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('tabs alignIcon start', () => {
    expect(
      create(
        <Tabs
          alignIcon={TabIconAlign.Start}
          onChange={tabClick}
          value={'tab1'}
          variant={TabVariant.pill}
        >
          {tabs.map((tab) => (
            <Tab key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('tabs alignIcon end', () => {
    expect(
      create(
        <Tabs
          alignIcon={TabIconAlign.End}
          onChange={tabClick}
          value={'tab1'}
          variant={TabVariant.pill}
        >
          {tabs.map((tab) => (
            <Tab key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('pill tabs large snapshot', () => {
    expect(
      create(
        <Tabs
          onChange={tabClick}
          size={TabSize.Large}
          value={'tab1'}
          variant={TabVariant.pill}
        >
          {tabs.map((tab) => (
            <Tab key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('pill tabs medium (default) snapshot', () => {
    expect(
      create(
        <Tabs onChange={tabClick} value={'tab1'} variant={TabVariant.pill}>
          {tabs.map((tab) => (
            <Tab key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('pill tabs medium (set) snapshot', () => {
    expect(
      create(
        <Tabs
          onChange={tabClick}
          size={TabSize.Medium}
          value={'tab1'}
          variant={TabVariant.pill}
        >
          {tabs.map((tab) => (
            <Tab key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('pill tabs small snapshot', () => {
    expect(
      create(
        <Tabs
          onChange={tabClick}
          size={TabSize.Small}
          value={'tab1'}
          variant={TabVariant.pill}
        >
          {tabs.map((tab) => (
            <Tab key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('stat tabs medium snapshot', () => {
    expect(
      create(
        <Tabs onChange={tabClick} value={'tab1'}>
          {tabs.map((tab) => (
            <Stat key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('stat tabs small snapshot', () => {
    expect(
      create(
        <Tabs onChange={tabClick} size={TabSize.Small} value={'tab1'}>
          {tabs.map((tab) => (
            <Stat key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('stat tabs xsmall snapshot', () => {
    expect(
      create(
        <Tabs onChange={tabClick} size={TabSize.XSmall} value={'tab1'}>
          {tabs.map((tab) => (
            <Stat key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('stat tabs medium vertical fullWidth snapshot', () => {
    expect(
      create(
        <Tabs onChange={tabClick} value={'tab1'} direction="vertical" fullWidth>
          {tabs.map((tab) => (
            <Stat key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('stat tabs medium horizontal maxWidth lineClamp snapshot', () => {
    expect(
      create(
        <Tabs
          onChange={tabClick}
          value={'tab1'}
          direction="vertical"
          maxWidth={240}
          lineClamp={2}
        >
          {tabs.map((tab) => (
            <Stat key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('stat tabs medium vertical snapshot', () => {
    expect(
      create(
        <Tabs onChange={tabClick} value={'tab1'} direction="vertical">
          {tabs.map((tab) => (
            <Stat key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('stat tabs small vertical snapshot', () => {
    expect(
      create(
        <Tabs
          onChange={tabClick}
          size={TabSize.Small}
          value={'tab1'}
          direction="vertical"
        >
          {tabs.map((tab) => (
            <Stat key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('stat tabs xsmall vertical snapshot', () => {
    expect(
      create(
        <Tabs
          onChange={tabClick}
          size={TabSize.XSmall}
          value={'tab1'}
          direction="vertical"
        >
          {tabs.map((tab) => (
            <Stat key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });

  test('stat tabs with buttons snapshot', () => {
    expect(
      create(
        <Tabs onChange={tabClick} size={TabSize.XSmall} value={'tab1'}>
          {tabsWithButtons.map((tab) => (
            <Stat key={tab.value} {...tab} />
          ))}
        </Tabs>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
