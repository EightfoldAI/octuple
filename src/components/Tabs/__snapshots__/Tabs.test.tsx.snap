// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel tabs snapshot 1`] = `
<div>
  <div
    className="tabWrap"
    role="tablist"
  >
    <button
      aria-label="Tab 1"
      aria-selected={true}
      className="tab active"
      onClick={[Function]}
      role="tab"
    >
      <span
        aria-hidden={false}
        className="icon iconWrapper"
        role="presentation"
      >
        <svg
          role="presentation"
          style={
            Object {
              "height": "20px",
              "width": "20px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
            style={
              Object {
                "fill": "currentColor",
              }
            }
          />
        </svg>
      </span>
      <span
        className="label"
      >
        Tab 1
      </span>
      <div
        className="tabIndicator"
        data-flip-config="{\\"translate\\":true,\\"scale\\":true,\\"opacity\\":true}"
        data-flip-id="tabIndicator"
        data-portal-key="portal"
      />
    </button>
    <button
      aria-label="Tab 2"
      aria-selected={false}
      className="tab"
      onClick={[Function]}
      role="tab"
    >
      <span
        aria-hidden={false}
        className="icon iconWrapper"
        role="presentation"
      >
        <svg
          role="presentation"
          style={
            Object {
              "height": "20px",
              "width": "20px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
            style={
              Object {
                "fill": "currentColor",
              }
            }
          />
        </svg>
      </span>
      <span
        className="label"
      >
        Tab 2
      </span>
    </button>
    <button
      aria-label="Tab 3"
      aria-selected={false}
      className="tab"
      onClick={[Function]}
      role="tab"
    >
      <span
        aria-hidden={false}
        className="icon iconWrapper"
        role="presentation"
      >
        <svg
          role="presentation"
          style={
            Object {
              "height": "20px",
              "width": "20px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
            style={
              Object {
                "fill": "currentColor",
              }
            }
          />
        </svg>
      </span>
      <span
        className="label"
      >
        Tab 3
      </span>
    </button>
    <button
      aria-label="Tab 4"
      aria-selected={false}
      className="tab"
      disabled={true}
      onClick={[Function]}
      role="tab"
    >
      <span
        aria-hidden={false}
        className="icon iconWrapper"
        role="presentation"
      >
        <svg
          role="presentation"
          style={
            Object {
              "height": "20px",
              "width": "20px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
            style={
              Object {
                "fill": "currentColor",
              }
            }
          />
        </svg>
      </span>
      <span
        className="label"
      >
        Tab 4
      </span>
    </button>
  </div>
</div>
`;
